packages/jest-transform/src/__tests__/ScriptTransformer.test.ts

```diff
@@ -2019,6 +2019,36 @@ describe('ScriptTransformer', () => {
     expect(fs.readFileSync).toHaveBeenCalledWith('/fruits/banana.js', 'utf8');
   });
 
+  it('regardless of sync/async, does not reuse the in-memory cache between different projects', async () => {
+    const scriptTransformer = await createScriptTransformer({
+      ...config,
+      transform: [['\\.js$', 'test_preprocessor', {}]],
+    });
+
+    scriptTransformer.transform('/fruits/banana.js', getCoverageOptions());
+
+    const yetAnotherScriptTransformer = await createScriptTransformer({
+      ...config,
+      transform: [['\\.js$', 'test_preprocessor', {}]],
+    });
+    yetAnotherScriptTransformer.transform(
+      '/fruits/banana.js',
+      getCoverageOptions(),
+    );
+
+    expect(fs.readFileSync).toHaveBeenCalledTimes(2);
```

Comment:
Is this expected behavior? From the title of the test it suggests not, but it does seem reasonable potentially to reuse the cache if the configString is exactly the same?